server/test/unit/server/db/test_connection.py

```diff
@@ -730,45 +730,48 @@ class TestDatabaseRetryOnInitialConnectionSupport(unittest.TestCase):
 
     @patch('pulp.server.db.connection._CONNECTION', None)
     @patch('pulp.server.db.connection._DATABASE', None)
+    @patch('pulp.server.db.connection.MongoEngineConnectionError', TestMongoEngineConnectionError)
     @patch('pulp.server.db.connection.mongoengine')
     @patch('pulp.server.db.connection.time.sleep', Mock())
     def test_retry_waits_when_mongoengine_connection_error_is_raised(self, mock_mongoengine):
         def break_out_on_second(*args, **kwargs):
             mock_mongoengine.connect.side_effect = StopIteration()
-            raise MongoEngineConnectionError()
+            raise TestMongoEngineConnectionError()
 
         mock_mongoengine.connect.side_effect = break_out_on_second
-        mock_mongoengine.connection.ConnectionError = MongoEngineConnectionError
+        mock_mongoengine.connection.MongoEngineConnectionError = TestMongoEngineConnectionError
```

Comment:
I think you do not need this line. It patches the `MongoEngineConnectionError` for a 0.11+ mongoengine but it is not needed at all since you patched it already in the namespace of `connection.py` on line 733.